RequestService/RequestService.Repo/Repository.cs

```diff
@@ -983,6 +991,58 @@ public GetRequestDetailsResponse GetRequestDetails(int requestID)
             return response;
         }
 
+        public int UpdateShiftStatusToAccepted(int requestID, HelpMyStreet.Utils.Enums.SupportActivities activity, int createdByUserID, int volunteerUserID, CancellationToken cancellationToken)
+        {
+            byte supportActivity = (byte)activity;
+            byte jobStatusOpen = (byte)JobStatuses.Open;
+            byte requestTypeShift = (byte)RequestType.Shift;
+
+            var job = _context.Job
+                .Include(i=> i.NewRequest)
+                .FirstOrDefault(x=> x.SupportActivityId == supportActivity
+                && x.RequestId == requestID
+                && x.JobStatusId == jobStatusOpen
+                && x.NewRequest.RequestType == requestTypeShift);
+
+            if(job != null)
+            {
+                job.JobStatusId = (byte)JobStatuses.Accepted;
+                job.VolunteerUserId = volunteerUserID;
+                AddJobStatus(job.Id, createdByUserID, volunteerUserID, (byte)JobStatuses.Accepted);
+                _context.SaveChanges();
+                return job.Id;
+            }
+            else
+            {
+                throw new Exception($"Unable to UpdateShiftStatus for RequestID { requestID}");
```

Comment:
This could just mean all vaccancies have been filled; we should ensure a suitable response is returned to the calling service.  (Should be the same as happens when attempting to accept an already-accepted request currently.)